components/dev-tools/easybuild/SPECS/easybuild.spec

```diff
@@ -48,25 +48,24 @@ you to manage (scientific) software on High Performance Computing (HPC)
 systems in an efficient way.
 
 %prep
-mkdir %{buildroot}
+%setup -q -n %{pname}-%{version} -a1 -a2 -a3
 
 %build
-
-cd %{buildroot}
-cp %{_sourcedir}/*py .
-
-export EASYBUILD_BOOTSTRAP_SOURCEPATH=%{_sourcedir}
-export EASYBUILD_INSTALLPATH=%{install_path}
-export EASYBUILD_MODULE_SYNTAX=Tcl
-export PATH=${LMOD_DIR}:${PATH}
-
-MODULEPATH= python3 ./bootstrap_eb.py %{buildroot}/%{install_path}
-
-rm bootstrap_eb.py*
-pushd %{buildroot}%{install_path}/modules/tools/EasyBuild/
-rm %version
-ln -s ../../all/EasyBuild/%version .
-popd
+%{__python3} setup.py build
+
+for eb in easyblocks easyconfigs framework; do
+	cd %{pname}-${eb}-%{version}
+	%{__python3} setup.py build
+	cd ..
+done
+
+%install
+%{__python3} setup.py install --prefix=%{install_path} --root=%{buildroot}
+for eb in easyblocks easyconfigs framework; do
+	cd %{pname}-${eb}-%{version}
+	%{__python3} setup.py install --prefix=%{install_path} --root=%{buildroot}
```

Comment:
If possible, use `pip install` instead:

```python
%{__python3} -m pip install --prefix=%{install_path} --root=%{buildroot} .
```

the trailing `.` is important here, you need to pass the path to the unpacked sources to `pip install`

Reply:
Never seen `pip install` in an RPM spec file before. Any reason you recommend this?

Reply:
`setup.py install` has been deprecated[^1][^2].

[^1]: https://github.com/pypa/setuptools/pull/2824/files
[^2]: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

Reply:
Indeed, exactly because of this: using `setup.py install` is deprecated.

We've also seen actual problems with recent setuptools versions when sticking to `setup.py install` (cfr. https://github.com/easybuilders/easybuild/issues/817#issuecomment-1281117533)